### PR TITLE
Generic Service in DataProvider

### PR DESCRIPTION
**Pull request by Alexees**
## Overview

Derivates of the BaseDataProvider, mostly all DeviceManagers, rely on their respective service.

The following changes are applied:
- The BaseDataProvider Service field is now generic, no casting needed anymore. The class is abstract anyways, which made the change easy.
- Removed the BaseInputDeviceManager's redundant InputSystem attribute as now it just can and always should have accessed the Service attribute

## Changes
- Fixes: #6778

### CODE REVIEW

File: Assets/MixedRealityToolkit/Services/BaseDataProvider.cs [hunk 1]
```diff
@@ -38,11 +40,16 @@ public abstract class BaseDataProvider : BaseService, IMixedRealityDataProvider
...
         protected BaseDataProvider(
-            IMixedRealityService service,
+            T service,
             string name = null,
             uint priority = DefaultPriority,
             BaseMixedRealityProfile profile = null) : base()
         {
+            if (service == null)
+            {
+                Debug.LogError($"{name} requires a valid input system instance.");
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit/Services/BaseDataProvider.cs]: error comment seems incorrect. BaseDataProvider can apply to other systems than Input system

**Comment by thalbern** [comment]
docs need fixing - triggered CI again for tests as they seemed to pass but failed publishing step

**Comment by keveleigh** [comment]
@Alexees I opened https://github.com/HoloSpaces/MixedRealityToolkit-Unity/pull/1 on your repo to resolve the build/docs/test failures mentioned above.

**Comment by keveleigh** [comment]
Looks like the validation failure is two `InputSystem` references in WindowsMixedRealityEyeGazeProvider.

**Pull request merged**